src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -42,15 +42,79 @@ public IISExpressDeployer(DeploymentParameters deploymentParameters, ILoggerFact
                 // Start timer
                 StartTimer();
 
-                // For now we always auto-publish. Otherwise we'll have to write our own local web.config for the HttpPlatformHandler
-                DotnetPublish();
+                var targetFramework = DeploymentParameters.TargetFramework;
 
-                var contentRoot = DeploymentParameters.PublishedApplicationRootPath;
+                // IISIntegration uses this layout
+                var dllRoot = Path.Combine(DeploymentParameters.ApplicationPath, "bin", DeploymentParameters.RuntimeArchitecture.ToString(),
+                    DeploymentParameters.Configuration, targetFramework);
+
+                if (!Directory.Exists(dllRoot))
```

Comment:
Add a private function which returns the found dll root and sets PublishApplicationBeforeDeployment.